common/is_ubuntu_1604_or_older.sh

```diff
@@ -0,0 +1,12 @@
+#!/bin/sh
+# A script to check whether this machine is running
+# Ubuntu 16.04 or older.
+
+# Check for lsb_release program
+command -v lsb_release > /dev/null 2>&1 || exit 1
+
+# Check we're running on Ubuntu
+if [ ! $(lsb_release -is) = Ubuntu ]; then exit 1; fi
+
+# Check we're running on Ubuntu 16.04 or earlier
+if [ $(lsb_release -rs | sed 's/\.//') -gt 1604 ]; then exit 1; fi
```

Comment:
Ubuntu 17 also has the userland header-only libi2c-dev https://packages.ubuntu.com/artful/all/libi2c-dev/filelist

Reply:
I feel that this will also effect Debian, Fedora et al so I feel what we **want** to test for is whether the version of i2c-tools is >= 4 but this is infuriatingly difficult - all the actual tools are so shitly written they don't have --version :/

Reply:
Ok, after looking at the source for i2cdetect, -V **is** implemented just not documented. Grepping this would definitely be a better test:

On Ubuntu 16 (x86_64 and ARM64):
``i2cdetect version 3.1.1``
On Ubuntu 18 (x86_64):
``i2cdetect version 4.0``